torch/serialization.py

```diff
@@ -453,14 +453,31 @@ def persistent_load(saved_id):
             raise RuntimeError("Unknown saved id type: %s" % saved_id[0])
 
     f_is_real_file = _is_real_file(f)
-    if f_is_real_file and f.tell() == 0:
+
+    def raise_load_err_msg(pattern, e):
+        if pattern in str(e):
+            msg = (str(e) + ". You can only torch.load from a file that is seekable." +
+                            " Please pre-load the data into a buffer like io.BytesIO and try to load from it instead.")
+            raise type(e)(msg)
+        else:
+            raise e
+
+    try:
+        f_tell_is_0 = True if f.tell() == 0 else False
```

Comment:
for this part, you can just do `f_tell_is_0 = (f.tell() == 0)` (I'm not sure if the parenthesis is needed)

Reply:
right, I can change it. I think with parenthesis looks more clear.